worker/worker.py

```diff
@@ -316,19 +316,20 @@ def main():
     )
     (options, args) = parser.parse_args()
 
-    if len(args) != 2:
-        # Try to read parameters from the the config file
-        username = config.get("login", "username")
-        password = config.get("login", "password", raw=True)
-        if len(username) != 0 and len(password) != 0:
-            args.extend([username, password])
-        else:
-            sys.stderr.write("{} [username] [password]\n".format(sys.argv[0]))
-            worker_exit()
+    username = config.get("login", "username")
+    password = config.get("login", "password", raw=True)
+
+    if len(args) == 2:
+        username = args[0]
+        password = args[1]
+    elif len(username) == 0 or len(password) == 0:
```

Comment:
@tomtor I replicated the old logic to use the config information for `len(args) != 2`, otherwise we can use it only for `len(args) == 0` with `elif len(args) != 0 or len(username) == 0 or len(password) == 0:`

Reply:
@tomtor updated to use the latter unambiguous logic.